### DISCUSSION TITLE
Add items to shopping list without amounts

### DISCUSSION THREAD

**Discussion by fcassata** [mealie-recipes/mealie#4051] [Q&A]
Is it possible to add only the food items to shopping lists?

**Comment by boc-the-git** [comment]
Please provide more info on what you're suggesting here.

**Comment by fcassata** [comment]
Please excuse the lack of clarity. See the attached screenshot. I would like only the text in bold, or the actual food items, to be added to the list, not the units or amounts. I sync this list to our home assistant shopping list, and the current implementation causes issues with duplicates.

![Screenshot 2024-08-19 at 09 40 53](https://github.com/user-attachments/assets/64823da8-97ce-4279-91b8-a3690c89e2b1)

**Reply by boc-the-git** [reply]
My gut instinct is that this isn't a change that really makes sense for Mealie.. it seems like your problem is in whatever your sync mechanism to HA is.
Maybe I'm wrong..

**Reply by fcassata** [reply]
The sync mechanism is just the official Mealie integration. But regardless, if we were to move to only using the list on the Mealie side, I would not want amounts or units on the list. As the chef and buyer of the household, it doesn't make sense to me, and it only makes the list harder to read as I need to "parse" it on the fly when shopping.

**Reply by dacmcbibs** [reply]
I think (at least for me) what would make the most sense would be to separate the quantities from the ingredients into separate fields. This way if you add multiple recipes to a shopping list you would  get the total amount of the ingredient required across all recipes instead multiple lines of the same ingredient one for each recipe... bonus points if the conversion is done so that it visually makes the most sense for the end user.... Example instead of :

- 1 cup mayo
- 3 tablespoons of pickled relish
- 1 Tablespoon honey
- 1/4 cup of pickled relish
- 1/2 cup mayo
- 1 teaspoon honey

you would get : 
- 1 1/2 cups mayo
-  7 Tablespoons of picked relish (or 1/2 + 3 tablespoons pickled relish) 
-  4 teaspoons honey (or 1 tablespoon and 1 teaspoon honey)

This would make list sorting easier and shopping more straight forward since you now know how much total of each ingredient is needed

**Reply by michael-genson** [reply]
>  you would get the total amount of the ingredient required across all recipes instead multiple lines of the same ingredient one for each recipe

That's how it works today. Ingredients with the same unit and food are merged

**Reply by fcassata** [reply]
> That's how it works today. Ingredients with the same unit and food are merged

That makes sense, kinda like scaling a recipe.

Any chance we'll see a setting to add only the food to shopping lists?

**Reply by boc-the-git** [reply]
Quoting myself here :see_no_evil:, but:
> My gut instinct is that this isn't a change that really makes sense for Mealie..

It seems like a very niche use case and I'm not sure there is any good reason to do it that way. It's easy enough to ignore the quantity that is there. No one will force you to buy 1kg of something because your shopping list says 1kg, despite you wanting to buy a different amount of it.

If you set the quantity of things on the shopping list to 0, Mealie won't display a quantity (go try it). That could easily be scripted.

**Reply by fcassata** [reply]
> It seems like a very niche use case and I'm not sure there is any good reason to do it that way.

I guess it depends on how people use shopping lists in general.

> No one will force you to buy 1kg of something

Of course not, but having the actual item to buy postfixed with what can in some instances be more text than the actual item, means having to parse the actual list when shopping. The food items won't align either, so for example for dyslexic people it becomes a hunt for what the line actually is. 

___

I just think it's bad UX in general not to give the user the option to get a simple shopping list, but having the system insist it be more feature rich than one might need or want, sacrificing readability.

**Reply by luismqoliveira** [reply]
I would like this feature too because I'm currently using the integration in Home Assistant to sync lists do Bring! and Bring! doesn't handle well the igredient's icons if the amount is there. This would work great if the amount is ignored. 
I agree with a configuration for those who want to show the amount and for those who don't.

**Reply by fcassata** [reply]
> I would like this feature too because I'm currently using the integration in Home Assistant to sync lists do Bring! and Bring! doesn't handle well the igredient's icons if the amount is there. This would work great if the amount is ignored. I agree with a configuration for those who want to show the amount and for those who don't.

I don't think they're interested in adding it, they dismissed the suggestion as you can see above.

**Reply by luismqoliveira** [reply]
That's a shame. If this was a config would be usable for anyone and each user could config per shopping list how ingredients are shown

**Reply by gordonbazeley** [reply]
@boc-the-git To add to this - I would find it really useful to be able to copy a list of just the ingredient names (without quantities & units) so that I can paste them straight into my supermarket online shopping website. 

I'm based in the UK and use [Sainsburys supermarket](https://www.sainsburys.co.uk/gol-ui/groceries) - I generally copy a list of ingredients that I've manually compiled into their "Search a list of items" feature and would like to do the same with a list compiled by Mealie. 

When I use a list from Mealie with quantities & units included the Sainsbury's search engine fails to find the ingredients (without the quantities & units the searches succeed). I suspect that other supermarket websites would similarly struggle.
